core-bundle/contao/library/Contao/StringUtil.php

```diff
@@ -200,23 +200,36 @@ public static function substrHtml($strString, $intNumberOfChars)
 	/**
 	 * Decode all entities
 	 *
-	 * @param mixed   $strString     The string to decode
+	 * @param string|array   $strString     The string to decode
 	 * @param integer $strQuoteStyle The quote style (defaults to ENT_QUOTES)
 	 * @param string  $strCharset    An optional charset
 	 *
-	 * @return string The decoded string
+	 * @return string|array The decoded string
 	 */
 	public static function decodeEntities($strString, $strQuoteStyle=ENT_QUOTES)
 	{
-		if ((string) $strString === '')
+		$replace = static function (&$value) use ($strQuoteStyle) {
+			if ((string) $value === '')
+			{
+				$value = '';
+				return;
+			}
+
+			$value = preg_replace('/(&#*\w+)[\x00-\x20]+;/i', '$1;', $value);
+			$value = preg_replace('/(&#x*)([0-9a-f]+);/i', '$1$2;', $value);
+			$value = html_entity_decode($value, $strQuoteStyle | ENT_SUBSTITUTE | ENT_HTML5, 'UTF-8');
```

Comment:
```suggestion
			if (\is_string($value))
			{
				$value = preg_replace('/(&#*\w+)[\x00-\x20]+;/i', '$1;', $value);
				$value = preg_replace('/(&#x*)([0-9a-f]+);/i', '$1$2;', $value);
				$value = html_entity_decode($value, $strQuoteStyle | ENT_SUBSTITUTE | ENT_HTML5, 'UTF-8');
			}
```

@ausi Do we not need the `is_string()` check here as well?

Reply:
I think we could also leave it as is and let it throw a type error if the parameter is not a string. But your version is consistent with `StringUtil::convertBasicEntities()` so that’s probably better 👍